src/Microsoft.AspNetCore.Server.HttpSys/HttpSysOptions.cs

```diff
@@ -87,6 +86,21 @@ public long RequestQueueLimit
         /// </summary>
         public TimeSpan ShutdownTimeout { get; set; } = TimeSpan.FromSeconds(5);
 
+        internal List<string> ServerAddresses { get; private set; }
+
+        /// <summary>
+        /// Bind to given IP addresses and ports.
+        /// </summary>
+        public void Listen(params string[] addresses)
```

Comment:
Do we want a different API here? Kestrel listens on `Endpoints` for example but I'm not sure how much more value that brings. Maybe strings are sufficient.